docs/src/quickstart.md

```diff
@@ -107,66 +166,130 @@ processing" communities.
 Because images are just arrays, some environments (e.g.,
 Juno or IJulia/Jupyter) will display numeric arrays as arrays using a text
 representation but will display 2d arrays that have `Colorant`
-elements as images.  You can "convert" in the following ways:
+elements as images.
+
+## Color conversions are construction/view
+
+Conversions between different `Colorant`s are straightforward:
+
+```@repl pixel
+RGB.(img_gray) # Gray => RGB
+Gray.(img_rgb) # RGB => Gray
+```
+
+!!! note
+    You'll see [broadcasting](https://docs.julialang.org/en/v1/manual/arrays/#Broadcasting-1)
+    semantics used in JuliaImages here and there, check the documentation if
+    you're not familiar with it.
 
-```jldoctest
-using Images
+Sometimes, to work with other packages, you'll need to convert a ``m \times n``
+`RGB` images to ``m \times n \times 3`` image and vice versa. `channelview` and
+`colorview` are designed for this purpose. For example:
 
-# "Convert" a numeric array to a `Colorant` (but grayscale) array
-A = rand(8, 8)
-img = Gray.(A)               # creates a copy of the data
-img = colorview(Gray, A)     # creates a "view" which "reinterprets" but does not copy data
+```@repl pixel
+img_CHW = channelview(img_rgb); # 3 * 2 * 2
+summary(img_CHW)
+img_HWC = permutedims(img_CHW, (2, 3, 1)); # 2 * 2 * 3
+summary(img_HWC)
+```
+
+```@repl pixel
+img_CHW = permutedims(img_HWC, (3, 1, 2)); # 3 * 2 * 2
+summary(img_CHW)
+img_rgb = colorview(RGB, img_CHW); # 2 * 2
+summary(img_rgb)
+```
 
-# RGB images
-A = rand(3, 8, 8)            # Float64 data (64 bits per color channel)
-img = colorview(RGB, A)      # encodes as a 2d RGB{Float64} array
-A = rand(N0f8, 3, 8, 8)      # uses only 8 bits per channel
-img = colorview(RGB, A)      # encodes as a 2d RGB{Float64} array
+!!! note
+    The reason we use CHW (i.e., channel-height-width) order instead of HWC
+    is that this provides a memory friendly indexing mechanisim for `Array`.
+    By default, in Julia the first index is also the fastest (i.e., has
+    adjacent storage in memory). For more details, please check [Access arrays in memory order, along columns](https://docs.julialang.org/en/v1/manual/performance-tips/#Access-arrays-in-memory-order,-along-columns-1)
```

Comment:
For more details, please refer to the performance tip: [Access arrays in memory order, along columns.](https://docs.julialang.org/en/v1/manual/performance-tips/#Access-arrays-in-memory-order,-along-columns-1)